examples/gno.land/p/demo/uuid/uuid.gno

```diff
@@ -0,0 +1,132 @@
+// to standard UUID format strings, making them compatible with systems
+// requiring UUIDs.
+// add import "gno.land/p/demo/entropy" to use the entropy package. For of the pseudo-random number generator.
+
+package uuid
+
+import (
+	"encoding/binary"
+	"std"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/entropy"
+	"gno.land/p/demo/ufmt"
+)
+
+const (
+	TimestampLength uint8  = 41
+	MachineIDLength uint64 = 10
+	SequenceLength  uint64 = 12
+	MaxSequence     uint16 = 1<<SequenceLength - 1
+	MaxTimestamp    uint64 = 1<<TimestampLength - 1
+	MaxMachineID    uint64 = 1<<MachineIDLength - 1
+
+	machineIDMoveLength = SequenceLength
+	timestampMoveLength = MachineIDLength + SequenceLength
+)
+
+// UUID struct for generating unique identifiers.
+type UUID struct {
+	MachineID     uint64
+	sequence      uint16
+	StartTime     int64
+	LastTimestamp int64
+	entropy       *entropy.Instance
+}
+
+func NewUUID() *UUID {
+	u := &UUID{
+		StartTime: time.Date(2008, 11, 10, 23, 0, 0, 0, time.UTC).Unix() * 1000,
+		// counter:   0,
+		entropy: entropy.New(),
+	}
+	u.SetMachineID()
+	return u
+}
+
+// ID generates
+func (u *UUID) ID() uint64 {
+	current := u.incrementEntropy()
+	elapsedTime := current - u.StartTime
+	u.LastTimestamp = current
+
+	if uint64(elapsedTime) < MaxTimestamp {
+		panic("Exceeding the time limit")
+	}
+
+	id := (uint64(elapsedTime) << timestampMoveLength) |
+		(uint64(u.MachineID) << machineIDMoveLength) |
+		uint64(u.incrementSequence())
+	return id
+}
+
+func (u *UUID) incrementSequence() uint16 {
+	u.sequence = (u.sequence + 1) & MaxSequence
+	return u.sequence
+}
+
+func (u *UUID) incrementEntropy() int64 {
+	return int64(u.entropy.Value())
+}
+
+// SetMachineID sets the machine ID based on the caller’s address.
+func (u *UUID) SetMachineID() {
+	caller := std.PrevRealm().Addr() // Retrieve the caller’s address
+	machineID := uint64(0)
+	for _, c := range caller.String() {
+		machineID += uint64(c)
+	}
+	machineID %= MaxMachineID
+	u.MachineID = machineID
+}
+
+func IDToUUID(id uint64) string {
+	bytes := make([]byte, 16)
+
+	// Copy transformed ID into the second half of the array
+	copy(bytes[8:], uint64ToBytes(id))
+
+	// Use bits from the Snowflake ID to generate the first half of the UUID
+	bytes[0] = byte(id >> 60)
+	bytes[1] = byte(id >> 52)
+	bytes[2] = byte(id >> 44)
+	bytes[3] = byte(id >> 36)
+	bytes[4] = byte(id >> 28)
+	bytes[5] = byte(id >> 20)
+	bytes[6] = byte(id >> 12)
+	bytes[7] = byte(id >> 4)
+
+	// set the version and variant bits according to UUID specification.
+	bytes[6] = (bytes[6] & 0x0f) | 0x40 // version 4 (random)
+	bytes[8] = (bytes[8] & 0x3f) | 0x80 // variant 1 (RFC 4122)
+
+	return ufmt.Sprintf("%s-%s-%s-%s-%s",
+		bytesToHex(bytes[0:4]),
+		bytesToHex(bytes[4:6]),
+		bytesToHex(bytes[6:8]),
+		bytesToHex(bytes[8:10]),
+		bytesToHex(bytes[10:16]))
+}
+
+func uint64ToBytes(i uint64) []byte {
+	buf := make([]byte, 8)
+	binary.BigEndian.PutUint64(buf, i)
+	return buf
+}
+
+func bytesToHex(bs []byte) string {
```

Comment:
What do you think about implementing this as:

`return "0x` + hex.EncodeToString(bs)`

?

Reply:
Thanks for the idea of ​​using `hex.EncodeToString()` I hadn't thought of it.
However I think it's better to stick to the standard generation of UUIDs because `0x` indicates that the value is in hexadecimal but is not used in UUID display standards. For the structure you have to keep the `-` to delimit the 5 groups and keep a quick visibility.

What do you think of this implementation?

```go
hexStr := hex.EncodeToString(bytes)
	return ufmt.Sprintf("%s-%s-%s-%s-%s",
		hexStr[0:8],
		hexStr[8:12],
		hexStr[12:16],
		hexStr[16:20],
		hexStr[20:],
	)
```

Reply:
Looks good